engine/src/test/java/org/camunda/bpm/engine/test/api/runtime/RestartProcessInstanceAsyncTest.java

```diff
@@ -1150,6 +1151,33 @@ public void shouldSetExecutionStartTimeInBatchAndHistory() {
     Assertions.assertThat(historicBatch.getExecutionStartTime()).isEqualToIgnoringMillis(TEST_DATE);
   }
 
+  @Test
+  public void shouldSetProcessInstanceAndDefinitionIdInHistoryJobLog() {
+    // given
+
+    ProcessDefinition processDefinition = testRule.deployAndGetDefinition(ProcessModels.TWO_TASKS_PROCESS);
+    ProcessInstance processInstance = runtimeService.createProcessInstanceById(processDefinition.getId())
+            .startBeforeActivity("userTask1")
+            .execute();
+    runtimeService.deleteProcessInstance(processInstance.getId(), "test");
+    Batch batch = runtimeService.restartProcessInstances(processDefinition.getId())
+            .startAfterActivity("userTask2")
+            .processInstanceIds(processInstance.getId())
+            .executeAsync();
+    helper.executeSeedJob(batch);
+    List<Job> executionJobs = helper.getExecutionJobs(batch);
+
+    // when
+    helper.executeJob(executionJobs.get(0));
+
+    // then
+    HistoricJobLog jobLog = historyService.createHistoricJobLogQuery().jobDefinitionType(Batch.TYPE_PROCESS_INSTANCE_RESTART).list().get(0);
+    System.out.println(jobLog);
+
+    assertEquals(processInstance.getProcessDefinitionId(), jobLog.getProcessDefinitionId());
+    assertEquals(processInstance.getRootProcessInstanceId(), jobLog.getProcessInstanceId());
```

Comment:
❌ The checks are performed for the historic job log, I would expect to check the jobs as well.